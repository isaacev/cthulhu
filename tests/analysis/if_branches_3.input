fn main () -> Void {
  if true {
    "abc";
  } else {
    "def"
  };
}
