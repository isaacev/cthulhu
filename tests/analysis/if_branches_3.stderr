
  INCOMPATIBLE IF AND ELSE TYPES  --------------------------------------------

  Expected `()` because of the return type of the if clause:

  2 |   if true {
    |           ^
  3 |     "abc";
    |     ^^^^^^
  4 |   } else {
    |   ^

  Expected the else clause to return a type compatible with `()`. Instead it
  returned `Str`:

  5 |     "def"
    |     ^^^^^

